dialog-polyfill.js

```diff
@@ -115,6 +121,26 @@
     },
 
     /**
+     * Maybe assign a default role attribute to this dialog that is being
+     * polyfilled. Without one, the dialog won't have a useful a11y tree generated
+     * for its core usage in a Web application.
+     */
+    maybeSetADefaultRoleValue: function(){
+      if ( !this.dialog.hasAttribute('role') || this.dialog.getAttribute('role') === ''  ) {
+        this.dialog.setAttribute( 'role', 'dialog'  );
```

Comment:
@samthor Are there use cases or examples of the dialog having value transitioning to not having a role attribute at all, making the MutationObserver continuously making sure there is one at *least* set to `dialog` undesirable?

w/ the native Dialog element it seems its tree doesn't actively remove such a role; though I'm not exactly too familiar with interpreting a11y trees. 

Perhaps @alice can help us out with this?